### PR TITLE
Accounting Climate Report PDF Generation

### PR DESCRIPTION
**Pull request by kallenilver**
Sample report as generate on my local server:

[Berge climate report - year.pdf](https://github.com/goclimate/goclimate/files/8553053/Berge.climate.report.-.year.pdf)

Format to enter other_emissions from admin panel
`{"accomodation": 2860, "bus": 1, "e_scooter": 6, "ferry": 283, "furniture": 171, "other_electronics": 908, "phone_plan": 80, "phone_tablet": 179, "public_transport": 4, "wearable": 227}`

### CODE REVIEW

File: app/controllers/business/accounting_emissions_calculations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+module Business
+  class AccountingEmissionsCalculationsController < ApplicationController
```
Comment (axelsson) [inline app/controllers/business/accounting_emissions_calculations_controller.rb:4]: Is it okay to leave these endpoints exposed for anyone to post/update/delete?

**Review by axelsson — CHANGES_REQUESTED** [review]
Good work! 💪 I don't really feel like I can give a decent review of the computations for the graphs and that stuff but I found some minor things and it left me with a few thoughts.

All of this feedback isn't necessarily specific to the changes in this PR but I just wanted to get it out somewhere, might be more relevant when implementing the business dashboard :) 
🟢 In the first table maybe we can write "-", "no data" or something when we're missing data
🟢 Maybe remove "purchased" from the names of the subcategories in "purchased goods"
🟢 It is very confusing that we're not using the same colors for the same categories in the different graphs
🟡  Maybe make it more clear of what period the report covers, when I see "Period: year,
Date of report: 2022-04-06", is it the last year from that date or the current year so far or the previous year? So maybe display the time period as an interval
🟡 If the table on page 5 is supposed to explain "other*" in all the charts, could we please make that more clear? Maybe by adding a small footnote on these pages with some explanatory text like "see a breakdown of the other emissions on page 5" or something? I find it very confusing as it is now
🟢 Should the "climate bookkeeper" be mentioned somewhere? Now I'm only seeing "business calculator" or "company calculator"

File: app/models/business/accounting_emissions_calculation.rb [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+module Business
+  class AccountingEmissionsCalculation < ApplicationRecord
+    belongs_to :organization
+    validates :calculation_period, inclusion: %w[year half-year quarter]
+    before_save :convert_other_emissions_to_json
+
+    CALCULATION_FIELDS = [
+      { name: 'energy' }.freeze,
+      { name: 'electricity_consumption_emissions', category: 'energy' }.freeze,
```
[the remark this answers is not in the document]
Comment (nanditagoclimate) [reply app/models/business/accounting_emissions_calculation.rb:11]: We need a JSON structured code like the above to be able to reuse the code from climate report PDF generation so enum won't be possible. In ruby `.freeze` works as a const declaration.

**Comment by nanditagoclimate** [comment]
The idea behind the first version of accounting PDF is to make it similar to the climate report PDF so we can pick up some of the comments in V2 definitely.
Climate report period will be added in V2 as well as at the moment we don't know the range(wither regular year or financial year)
Since we use the same underlying business calculator we have those wordings in but I leave it up to @kallenilver to decide if we need to add `Climate Book keeper` term anywhere in the report.

**Pull request merged**